plugin-api/src/main/java/org/sonarsource/sonarlint/plugin/api/issue/NewInputFileEdit.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * SonarLint Plugin API
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.plugin.api.issue;
+
+import org.sonar.api.batch.fs.InputFile;
+
+/**
+ * Describe a file edit for a {@link NewQuickFix} as a collection of {@link NewTextEdit}s on a given {@link InputFile}.
+ * Text edits are applied in the order they are added, insofar that their ranges do not overlap.
```

Comment:
In this case, would we throw ? Ignore the addition ? If we talk about this corner case, Iwe should be more precise about how we handle it

Reply:
I believe that the check should be made only when trying to apply a fix: doing it in advance would probably hurt performance, and we wouldn't have knowledge about any change that might have happened concurrently in the files.